src/Adapters/Files/GridFSBucketAdapter.js

```diff
@@ -68,26 +72,71 @@ export class GridFSBucketAdapter extends FilesAdapter {
     const stream = await bucket.openUploadStream(filename, {
       metadata: options.metadata,
     });
-    if (this._encryptionKey !== null) {
-      try {
-        const iv = crypto.randomBytes(16);
-        const cipher = crypto.createCipheriv(this._algorithm, this._encryptionKey, iv);
-        const encryptedResult = Buffer.concat([
-          cipher.update(data),
-          cipher.final(),
-          iv,
-          cipher.getAuthTag(),
-        ]);
-        await stream.write(encryptedResult);
-      } catch (err) {
-        return new Promise((resolve, reject) => {
-          return reject(err);
-        });
+
+    const iv = crypto.randomBytes(16);
```

Comment:
This line will be executed in every case, which can be unnecessary. Could you refactor like in the original code?

Reply:
If its the original order I think I would have to have duplicate initializations for `cipher` and `iv` when its a blob case and when else. I could also do the null ternary like I do with cipher, if you would prefer that?